src/map_inner.rs

```diff
@@ -0,0 +1,1142 @@
+#![allow(clippy::indexing_slicing)] // indexing will never panic here, because all of the index has been mod by the length of the vector.
+
+use std::collections::hash_map::RandomState;
+
+use utilities::{Cast, OverflowArithmetic};
+
+use std::hash::{BuildHasher, Hash, Hasher};
+
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use super::pointer::{AtomicPtr, SharedPtr};
+
+// Re-export `crossbeam_epoch::pin()` and `crossbeam_epoch::Guard`.
+pub use crossbeam_epoch::{pin, Guard};
+
+/// `KVPair` contains the key-value pair.
+#[derive(Debug)]
+pub struct KVPair<K, V> {
```

Comment:
should `KVPair` be `pub(crate)`?

Reply:
Since this module is private, use `pub(crate`) is not necessary. see: https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pub_crate